server/src/main/java/org/opensearch/index/translog/transfer/TranslogCheckpointTransferSnapshot.java

```diff
@@ -164,7 +173,8 @@ public TranslogCheckpointTransferSnapshot build() throws IOException {
                         checkpointGeneration,
                         minTranslogGeneration,
                         checkpointPath,
-                        reader.getCheckpointChecksum()
+                        reader.getCheckpointChecksum(),
+                        checkpointBytes
```

Comment:
Why do we need to read the bytes beforehand? Can we read them when actually required?